src/createSettingsFileTransformer.js

```diff
@@ -0,0 +1,162 @@
+  return isString(str) && str.indexOf('[') !== -1 && str.indexOf(']') !== -1;
+}
+function sanitizeArrayKey(pathStrSegment) {
+  return pathStrSegment.substr(
+    0,
+    // the name goes up to the start of the array bracket: 'someArrayName[]'
+    pathStrSegment.indexOf('[')
+  );
+}
+
+function isObject(value) {
+  if (value == null) {
+    return false;
+  }
+
+  return Boolean(
+    typeof value === 'object' &&
+      Object.prototype.toString.call(value) === '[object Object]' &&
+      !Array.isArray(value)
+  );
+}
+
+function isString(value) {
+  var type = typeof value;
+  return Boolean(
+    type === 'string' ||
+      (type === 'object' &&
+        value != null &&
+        !Array.isArray(value) &&
+        Object.prototype.toString.call(value) === '[object String]')
+  );
+}
+
+function traverseIntoSettings(
+  filePathString,
+  settings,
+  decorateWithDynamicHost
+) {
+  // nothing to do
+  if (
+    !isString(filePathString) ||
+    !filePathString.length ||
+    !isObject(settings)
+  ) {
+    return;
+  }
+
+  // get the next key to observe
+  var pathSegments = filePathString.split('.');
+  var settingToReplace = pathSegments.pop();
+  if (!settingToReplace) {
+    var error = new Error();
+    error.code = 'malformed-path-string';
+    throw error;
+  }
+
+  // base case
+  if (
+    settings.hasOwnProperty(settingToReplace) &&
+    !pathSegments.length &&
+    isString(settings[settingToReplace])
+  ) {
+    settings[settingToReplace] = decorateWithDynamicHost(
+      settings[settingToReplace]
+    );
+    return;
+  }
+
+  // still more work to do
+  if (pathSegments.length) {
+    var currentKey = pathSegments.shift();
+    // the next layer down will need the setting to replace back on the end
+    var remainingPathString = pathSegments.concat(settingToReplace).join('.');
+
+    if (isArrayReference(currentKey)) {
+      // 'someArrayReference[]' --> 'someArrayReference'
+      currentKey = sanitizeArrayKey(currentKey);
+      var settingsValue = settings[currentKey];
+      if (Array.isArray(settingsValue)) {
+        settingsValue.forEach(function (arrayEntryObject) {
+          return traverseIntoSettings(
+            remainingPathString,
+            arrayEntryObject,
+            decorateWithDynamicHost
+          );
+        });
+      }
+    } else {
+      return traverseIntoSettings(
+        remainingPathString,
+        settings[currentKey],
+        decorateWithDynamicHost
+      );
+    }
+  }
+}
+
+/**
+ * Returns a function that when called will attempt to traverse the passed in
+ * settings object to each file path and transform a relative URL to an absolute
+ * URL.
+ *
+ * @param isDynamicEnforced
+ * @param decorateWithDynamicHost
+ * @returns {(function(*=, *=, *=): (*))|*}
+ */
+module.exports = function (isDynamicEnforced, decorateWithDynamicHost) {
+  return function (settings, filePaths, moduleReferencePath) {
+    if (
+      !isDynamicEnforced ||
+      !isObject(settings) ||
```

Comment:
The top level settings is always a object. At least this is what I see in the turbine schema here: https://github.com/adobe/reactor-turbine-schemas/blob/master/schemas/container.json#L30-L32. I think it's safe to not check here if settings is an object.